lib/SILOptimizer/Utils/PerformanceInlinerUtils.cpp

```diff
@@ -732,12 +732,14 @@ SILFunction *swift::getEligibleFunction(FullApplySite AI,
   // A non-fragile function may not be inlined into a fragile function.
   if (Caller->isSerialized() &&
       !Callee->hasValidLinkageForFragileInline()) {
+#if 0
     if (!Callee->hasValidLinkageForFragileRef()) {
```

Comment:
Please don't remove this check. It catches cases where serializable functions reference non-public things.

Reply:
It is only for now, to reproduce the failure.

Reply:
BTW, this is what fails due to public_external [serialized] functions as I explained in chat.

Reply:
I have a public_external function from the stdlib which was imported. It calls another public_external function. Both are public_external [serialized]. We specialize the callee and give it a shared linkage. And we don't mark it [serialized]. Now we have a public_external [serialized] function that calls a shared non-serialized function. Later on, we assert in the code snippet above that I disabled temporarily.

Reply:
@slavapestov What should be the right fix for such a case in your opinion?

Reply:
> Please don't remove this check. It catches cases where serializable functions reference non-public things.

BTW, don't we catch it in SILVerifier anyways?

Reply:
The failure here https://ci.swift.org/job/swift-PR-Linux-smoke-test/1717/ seems to run into a the manifestation of the same issue. We specialize and replace the apply in the [serialized] caller function to call the shared non-serialized specialized callee function.

Reply:
Specializations of public serialized functions should always be shared serialized.

Reply:
And yes we will ultimately catch it in the verifier but it's better to catch it sooner, closer to the original problem.